packages/react/src/RendererContext.tsx

```diff
@@ -28,6 +28,8 @@ const RendererContext = React.createContext<GriffelRenderer>(createDOMRenderer()
  * @public
  */
 export const RendererProvider: React.FC<RendererProviderProps> = ({ children, renderer, targetDocument }) => {
+  'use no memo';
```

Comment:
We should probably address the underlying issue but I've opted to simply disable for now to keep this PR focused on landing the lint rule, not fixing anything it surfaces.

Have a first draft of a potential fix here, though I think it needs work: #566 566